Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -397,67 +346,110 @@ def _foregroundWidths(self):
-            for f in inputFiles:
-                # Flatten input files into a single list
-                if isinstance(f, str):
-                    flattened.append(f)
-                else:
-                    # f is a list; concatenate.
-                    flattened += f
-            beamCentre = self.getProperty(Prop.BEAM_CENTRE).value
-            beamAngle = self.getProperty(Prop.BEAM_ANGLE).value
-            dbPosWS = ''
-            if not self.getProperty(Prop.DIRECT_BEAM_POS_WS).isDefault:
-                dbPosWS = self.getProperty(Prop.DIRECT_BEAM_POS_WS).value
-            filename = flattened.pop(0)
-            numor = os.path.basename(filename).split('.')[0]
-            firstWSName = self._names.withSuffix('raw-' + numor)
-            beamPosWSName = self._names.withSuffix('beamPos-' + numor)
-            LoadILLReflectometry(Filename=filename,
-                                 OutputWorkspace=firstWSName,
-                                 DirectBeamPosition=dbPosWS,
-                                 BeamCentre=beamCentre,
-                                 BraggAngle=beamAngle,
-                                 XUnit='TimeOfFlight',
-                                 OutputBeamPosition=beamPosWSName,
-                                 EnableLogging=self._subalgLogging)
-            mergedWS = mtd[firstWSName]
-            beamPosWS = mtd[beamPosWSName]
-            self._cleanup.cleanupLater(beamPosWS)
+        """Return a raw input workspace."""
+        inputFiles = self.getPropertyValue(Prop.RUN)
+        inputFiles = inputFiles.replace(',', '+')
+        if inputFiles:
             mergedWSName = self._names.withSuffix('merged')
-            for i, filename in enumerate(flattened):
-                numor = os.path.basename(filename).split('.')[0]
-                rawWSName = self._names.withSuffix('raw-' + numor)
-                LoadILLReflectometry(Filename=filename,
-                                     OutputWorkspace=rawWSName,
-                                     DirectBeamPosition=dbPosWS,
-                                     XUnit='TimeOfFlight',
-                                     EnableLogging=self._subalgLogging)
-                rawWS = mtd[rawWSName]
-                mergedWS = MergeRuns(InputWorkspaces=[mergedWS, rawWS],
-                                     OutputWorkspace=mergedWSName,
-                                     EnableLogging=self._subalgLogging)
-                if i == 0:
-                    self._cleanup.cleanup(firstWSName)
-                self._cleanup.cleanup(rawWS)
-            return mergedWS, beamPosWS
-        ws = self.getProperty(Prop.INPUT_WS).value
-        self._cleanup.protect(ws)
-        if not self.getProperty(Prop.BEAM_POS_WS).isDefault:
-            beamPosWS = self.getProperty(Prop.BEAM_POS_WS).value
-            self._cleanup.protect(beamPosWS)
+            loadOption = {'XUnit': 'TimeOfFlight',
+                          'BeamCentre': 0.0}
```

Comment:
We should also pass the user given 2theta angle to the loader which will then write the value to the sample logs.

Reply:
This is a very good idea!